src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -504,9 +505,17 @@ private TimingFlowNodeStorage createStorage() throws IOException {
         FlowNodeStorage wrappedStorage;
 
         FlowDurabilityHint hint = getDurabilityHint();
-        wrappedStorage = (hint.isPersistWithEveryStep())
-                ? new SimpleXStreamFlowNodeStorage(this, getStorageDir())
-                : new BulkFlowNodeStorage(this, getStorageDir());
+        if (hint.isPersistWithEveryStep()) {
+            wrappedStorage = new SimpleXStreamFlowNodeStorage(this, getStorageDir());
+            wrappedStorage.setAvoidAtomicWrite(!hint.isAtomicWrite());
+            wrappedStorage.setAvoidForce(!hint.isForce());
+        } else {
+            wrappedStorage = new BulkFlowNodeStorage(this, getStorageDir());
+            /*
+             * avoidAtomicWrite and avoidForce are always true for BulkFlowNodeStorage; see its
+             * constructor for details.
+             */
+        }
```

Comment:
Discovered this preëxisting bug while testing this series of changes. While we deal with calling `setAvoidAtomicWrite` (and, as of this PR, `setAvoidForce`) for the wrapping `TimingFlowNodeStorage` returned by this method earlier on line 402, nothing ever dealt with calling `setAvoidAtomicWrite` (and, as of this PR, `setAvoidForce`) on the wrapped `SimpleXStreamFlowNodeStorage` or `BulkFlowNodeStorage` being created here. During my testing, I found that some code was calling the wrapped storage and not the wrapper, and that call wasn't respecting the durability settings. This hunk fixes that problem.